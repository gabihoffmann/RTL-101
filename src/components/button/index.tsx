interface ButtonProps {
  children?: React.ReactNode;
  label: string;
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}
export function Button({ label, onClick }: Readonly<ButtonProps>) {
  return (
    <button onClick={onClick}>
      <strong>{label}</strong>
    </button>
  );
}
