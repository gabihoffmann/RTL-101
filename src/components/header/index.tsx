import { useState } from "react";

export function Header() {
  const [count, setCount] = useState(0);
  return (
    <div>
      <h1>Title</h1>
      <h2>Subtitle</h2>

      <div>
        <span>Counter: {count}</span>
        <button onClick={() => setCount((old) => ++old)}>add</button>
      </div>
    </div>
  );
}
