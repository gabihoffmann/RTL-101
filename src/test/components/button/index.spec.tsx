import { describe, expect, it, vitest } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";

import { Button } from "../../../components/button";

describe("Test the button successfully", () => {
  it("renders the button with success", () => {
    const buttonText = "Login";
    const { container } = render(<Button label={buttonText} />);
    const buttonElement = screen.getByRole("button", { name: buttonText });

    expect(buttonElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("should be call the function onClick from button with success", () => {
    const buttonText = "Login";
    const onClick = vitest.fn();

    render(<Button label={buttonText} onClick={onClick} />);
    const buttonElement = screen.getByRole("button", { name: buttonText });
    fireEvent.click(buttonElement);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
