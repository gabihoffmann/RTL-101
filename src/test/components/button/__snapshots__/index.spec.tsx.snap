// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test the button successfully > renders the button with success 1`] = `
<div>
  <button>
    <strong>
      Login
    </strong>
  </button>
</div>
`;
